### PR TITLE
[INLONG-9788][Sort] Supports data parse that contains delimiters in kv and csv data format

### LINKED ISSUE #9788: [Feature][Sort] Supports data parse that contains delimiters in kv and csv data content
### Description

kv format：
        String kvString1 = "name=n&age=10";
        Map<String, String> map1 = StringUtils.splitKv(kvString1, '&',
                '=', '\\', '\'');
        assertEquals("n", map1.get("name"));
        assertEquals("10", map1.get("age"));

        String kvString2 = "name=&age=20&";
        Map<String, String> map2 = StringUtils.splitKv(kvString2, '&',
                '=', '\\', '\'');
        assertEquals("", map2.get("name"));
        assertEquals("20&", map2.get("age"));

        String kvString3 = "name==&age=20&&&value=aaa&dddd&";
        Map<String, String> map3 = StringUtils.splitKv(kvString3, '&',
                '=', '\\', '\'');
        assertEquals("=", map3.get("name"));
        assertEquals("20&&", map3.get("age"));
        assertEquals("aaa&dddd&", map3.get("value"));

        String kvString4 = "name==&age=20&&\nname1==&age1=20&&";
        List<Map<String, String>> map4 = StringUtils.splitKv(kvString4, '&',
                '=', '\\', '\'', '\n');
        assertEquals("=", map4.get(0).get("name"));
        assertEquals("20&&", map4.get(0).get("age"));
        assertEquals("=", map4.get(0).get("name1"));
        assertEquals("20&&", map4.get(0).get("age1"));

        String kvString5 = "name==&age=20&&\nname1==&age1=20&&&value=aaa&dddd&";
        List<Map<String, String>> map5 = StringUtils.splitKv(kvString5, '&',
                '=', '\\', '\'', '\n');
        assertEquals("=", map5.get(0).get("name"));
        assertEquals("20&&", map5.get(0).get("age"));
        assertEquals("=", map5.get(0).get("name1"));
        assertEquals("20&&", map5.get(0).get("age1"));
        assertEquals("aaa&dddd&", map5.get(0).get("value"));

        String kvString6 = "name==&age=20&&\\";
        List<Map<String, String>> map6 = StringUtils.splitKv(kvString6, '&',
                '=', '\\', '\'', '\n');
        assertEquals("=", map6.get(0).get("name"));
        assertEquals("20&&", map6.get(0).get("age"));

        String kvString7 = "name==&age=20&&'";
        List<Map<String, String>> map7 = StringUtils.splitKv(kvString7, '&',
                '=', '\\', '\'', '\n');
        assertEquals("=", map7.get(0).get("name"));
        assertEquals("20&&", map7.get(0).get("age"));

        String kvString8 = "name=\\=&age=20&a&'";
        List<Map<String, String>> map8 = StringUtils.splitKv(kvString8, '&',
                '=', '\\', '\'', '\n');
        assertEquals("=", map8.get(0).get("name"));
        assertEquals("20&a&", map8.get(0).get("age"));

        String kvString9 = "name=\\=&age=20&a\\&'";
        List<Map<String, String>> map9 = StringUtils.splitKv(kvString9, '&',
                '=', '\\', '\'', '\n');
        assertEquals("=", map8.get(0).get("name"));
        assertEquals("20&a&", map8.get(0).get("age"));

csv format：
        String csvString1 = "name|age=20\\||&'";
        String[][] csv1Array1 = StringUtils.splitCsv(csvString1, '|',
                '\\', '\'', '\n');

        assertEquals("age=20|", csv1Array1[0][1]);
        assertEquals("&", csv1Array1[0][2]);

        String csvString2 = "name|age=20\\||&'\n\name|age=20\\||&'\n\n|home|\\home\\";
        String[][] csv1Array2 = StringUtils.splitCsv(csvString2, '|',
                '\\', '\'', '\n');

        assertEquals("name", csv1Array2[0][0]);
        assertEquals("age=20|", csv1Array2[0][1]);
        assertEquals("&\n\name|age=20\\||&", csv1Array2[0][2]);
        assertEquals("", csv1Array2[2][0]);
        assertEquals("home", csv1Array2[2][1]);
        assertEquals("home", csv1Array2[2][2]);

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
- Fixes #9788

### CODE REVIEW

File: inlong-sort/sort-formats/format-common/src/main/java/org/apache/inlong/sort/formats/util/StringUtils.java [hunk 1]
```diff
@@ -450,11 +489,11 @@ public static String[][] splitCsv(
...
-
-            case STATE_ESCAPING:
-                throw new IllegalArgumentException(String.format("Not closed escaping. Text=[%s].", text));
-            case STATE_QUOTING:
-                throw new IllegalArgumentException(String.format("Not closed quoting. Text=[%s].", text));
+//
+//            case STATE_ESCAPING:
+//                throw new IllegalArgumentException(String.format("Not closed escaping. Text=[%s].", text));
+//            case STATE_QUOTING:
+//                throw new IllegalArgumentException(String.format("Not closed quoting. Text=[%s].", text));
```
Comment (aloyszhang) [inline inlong-sort/sort-formats/format-common/src/main/java/org/apache/inlong/sort/formats/util/StringUtils.java:496]: It's better to remove these lines of code if they are not used.

File: inlong-sort/sort-formats/format-common/src/test/java/org/apache/inlong/sort/formats/common/StringUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package org.apache.inlong.sort.formats.common;
```
Comment (aloyszhang) [inline inlong-sort/sort-formats/format-common/src/test/java/org/apache/inlong/sort/formats/common/StringUtilsTest.java:1]: Missing Apache license header.

**Comment by aloyszhang** [comment]
I found that we already have several tests for `splitKv` or `splitCsv`
https://github.com/apache/inlong/blob/master/inlong-sort/sort-formats/format-rowdata/format-rowdata-kv/src/test/java/org/apache/inlong/sort/formats/kv/KvUtilsTest.java
https://github.com/apache/inlong/blob/master/inlong-sort/sort-formats/format-row/format-kv/src/test/java/org/apache/inlong/sort/formats/kv/KvUtilsTest.java
https://github.com/apache/inlong/blob/master/inlong-sort/sort-formats/format-row/format-csv/src/test/java/org/apache/inlong/sort/formats/csv/CsvUtilsTest.java

Is it necessacery to add a new test class?

**Comment by baomingyu** [comment]
> I found that we already have several tests for `splitKv` or `splitCsv`
> 
> https://github.com/apache/inlong/blob/master/inlong-sort/sort-formats/format-rowdata/format-rowdata-kv/src/test/java/org/apache/inlong/sort/formats/kv/KvUtilsTest.java
> 
> https://github.com/apache/inlong/blob/master/inlong-sort/sort-formats/format-row/format-kv/src/test/java/org/apache/inlong/sort/formats/kv/KvUtilsTest.java
> 
> https://github.com/apache/inlong/blob/master/inlong-sort/sort-formats/format-row/format-csv/src/test/java/org/apache/inlong/sort/formats/csv/CsvUtilsTest.java
> 
> Is it necessacery to add a new test class?

All existing tests are tests of data in kv and csv formats that combine expected rules. The new ones are scenarios that can parse data in kv and csv formats that do not comply with the rules.

**Comment by aloyszhang** [comment]
> All existing tests are tests of data in kv and csv formats that combine expected rules. The new ones are scenarios that can parse data in kv and csv formats that do not comply with the rules.

What I am concerned about is whether the new test cases can be added to the existing UT class(the three UT classes mentioned above) or not.

File: inlong-sort/sort-formats/format-common/src/main/java/org/apache/inlong/sort/formats/util/StringUtils.java (hunk 1 above)
Comment (baomingyu) [reply inlong-sort/sort-formats/format-common/src/main/java/org/apache/inlong/sort/formats/util/StringUtils.java]: done

**Comment by baomingyu** [comment]
> > All existing tests are tests of data in kv and csv formats that combine expected rules. The new ones are scenarios that can parse data in kv and csv formats that do not comply with the rules.
> 
> What I am concerned about is whether the new test cases can be added to the existing UT class(the three UT classes mentioned above) or not.
> 
> Or, are there some reasons that we must put the new test cases into a new UT class?

New tests test new scenarios，whether to put it in the inventory will be adjusted later.

**Comment by aloyszhang** [comment]
> New tests test new scenarios，whether to put it in the inventory will be adjusted later.

Make sense to me. We can unify the UT later.

**Review by aloyszhang — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**